tests/test.py

```diff
@@ -0,0 +1,192 @@
+def test_conditional_operator(input):
+    patch = "conditional_operator.cocci"
+    result = compare(patch, input, 0)
+    assert(result == True)
+    
+@pytest.mark.parametrize("input", patch_to_input_mapping["implicit_predicate.cocci"])
+def test_implicit_predicate(input):
+    patch = "implicit_predicate.cocci"
+    result = compare(patch, input, 0)
+    assert(result == True)
+
+@pytest.mark.parametrize("input", patch_to_input_mapping["logic_as_controlflow.cocci"])
+def test_logic_as_controlflow(input):
+    patch = "logic_as_controlflow.cocci"
+    result = compare(patch, input, 0)
+    assert(result == True)
+
+@pytest.mark.parametrize("input", patch_to_input_mapping["omitted_curly_braces.cocci"])
+def test_omitted_curly_braces(input):
+    patch = "omitted_curly_braces.cocci"
+    result = compare(patch, input, 0)
+    assert(result == True)
+
+@pytest.mark.parametrize("input", patch_to_input_mapping["post_incdec.cocci"])
+def test_post_incdec(input):
+    patch = "post_incdec.cocci"
+    result = compare(patch, input, 0)
+    assert(result == True)
+
+@pytest.mark.parametrize("input", patch_to_input_mapping["pre_incdec.cocci"])
+def test_pre_incdec(input):
+    patch = "pre_incdec.cocci"
+    result = compare(patch, input, 0)
+    assert(result == True)
+
+@pytest.mark.parametrize("input", patch_to_input_mapping["repurposed_variable.cocci"])
+def test_repurposed_variable(input):
+    patch = "repurposed_variable.cocci"
+    result = compare(patch, input, 0)
+    assert(result == True)
+
+@pytest.mark.parametrize("input", patch_to_input_mapping["reversed_subscripts.cocci"])
+def test_reversed_subscripts(input):
+    patch = "reversed_subscripts.cocci"
+    result = compare(patch, input, 0)
+    assert(result == True)
+
+@pytest.mark.parametrize("input", patch_to_input_mapping["type_conversion.cocci"])
+def test_type_conversion(input):
+    patch = "type_conversion.cocci"
+    result = compare(patch, input, 0)
+    assert(result == True)
+
+### Modify this later
+patch_to_test_mapping = {
+    "assignment_as_value.cocci": "test_assignment_as_value",
+    "change_of_literal_encoding.cocci": "test_change_of_literal_encoding",
+    "comma_operator.cocci": "test_comma_operator",
+    "conditional_operator.cocci": "test_conditional_operator",
+    "implicit_predicate.cocci": "test_implicit_predicate",
+    "logic_as_controlflow.cocci": "test_logic_as_controlflow",
+    "omitted_curly_braces.cocci": "test_omitted_curly_braces",
+    "post_incdec.cocci": "test_post_incdec",
+    "pre_incdec.cocci": "test_pre_incdec",
+    # "repurposed_variable.cocci": "test_repurposed_variable",
+    # "reversed_subscripts.cocci": "test_reversed_subscripts",
+    # "type_conversion.cocci": "test_type_conversion",
+}
+
+# patch file list from keyboard selection, call corresponding test functions (same name as patch) through args, append input files in for each pytest parameter,  
```

Comment:
Tests should not be interactive. If they are interactive, you won't be able to run them on CI. Even if you want to keep that feature, you can (and should) define a function which has a path as an input parameter and then test that function. If you absolutely have to test some code which relies on user's input (which is not the case here), you can mock those interactions.